File: src/actions/java/searchWords.java
import com.cosmian.jna.findex.Findex;
//
// Search a list of words
//
public static Map<Keyword, Set<Location>> searchWords(byte[] key, byte[] label, Set<Keyword> keywords, Database db) {
  return Findex.search(key, label, keywords, db)
}

File: src/actions/javascript/searchWords.ts
import { FetchChains, FetchEntries, Findex, FindexKey, Label, Location } from "cloudproof_js";

//
// Search a list of words
//
export const searchWords = async (
  masterKey: FindexKey,
  label: Label,
  words: string[],
  fetchEntries: FetchEntries,
  fetchChains: FetchChains
): Promise<Location[]> => {
  const { search } = await Findex();

  const results = await search(masterKey, label, words, fetchEntries, fetchChains);

  return results.locations();
};
